### PR TITLE
Don't upload draft if PgpMessageBuilder is used

### PR DESCRIPTION
**Pull request by Valodim**
also, more tests. Those aren't very pretty I'm afraid, due to the whole callback deal of the OpenPgpApi :(

### CODE REVIEW

**Comment by ArchangeGabriel** [comment]
Not sure what that does mean: will drafts of to-be-send-encrypted messages not be saved at all? If so, that doesn’t look like a good solution.

I would rather have what Thunderbird does here: saving drafts encrypted to my key (just like sent messages are, right?).

**Comment by Valodim** [comment]
This is a short term solution. It will probably be a while until I get to doing this right and save drafts encrypted, but as always - PRs welcome.

**Comment by ArchangeGabriel** [comment]
Make sense, thanks. ;)

**Comment by ArchangeGabriel** [comment]
P.S.: Note that your commit title is misleading for the same reason.

File: k9mail/src/test/java/org/openintents/openpgp/util/ShadowOpenPgpAsyncTask.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+import android.os.AsyncTask;
+
+import org.robolectric.annotation.Implementation;
+import org.robolectric.annotation.Implements;
+import org.robolectric.annotation.RealObject;
+import org.robolectric.shadows.ShadowAsyncTask;
+
+
+@Implements(OpenPgpApi.OpenPgpAsyncTask.class)
+public class ShadowOpenPgpAsyncTask extends ShadowAsyncTask<Void, Integer, Intent> {
```
Comment (cketti) [inline k9mail/src/test/java/org/openintents/openpgp/util/ShadowOpenPgpAsyncTask.java]: I've never looked closely into Robolectric's shadow classes. What does this do and why is it necessary?

File: k9mail/src/test/java/com/fsck/k9/activity/compose/RecipientPresenterTest.java [hunk 2]
```diff
@@ -30,19 +44,20 @@
 
 
 @RunWith(RobolectricTestRunner.class)
-@Config(manifest = "src/main/AndroidManifest.xml", sdk = 21)
+@Config(manifest = "src/main/AndroidManifest.xml", sdk = 21, shadows={ShadowOpenPgpAsyncTask.class})
```
Comment (cketti) [inline k9mail/src/test/java/com/fsck/k9/activity/compose/RecipientPresenterTest.java]: Code style

File: k9mail/src/test/java/com/fsck/k9/activity/compose/RecipientPresenterTest.java [hunk 3]
```diff
@@ -30,19 +44,20 @@
...
 public class RecipientPresenterTest {
-    public static final ReplyToAddresses TO_ADDRESSES = new ReplyToAddresses(Address.parse("to@example.org"));
-    public static final List<Address> ALL_TO_ADDRESSES = Arrays.asList(Address.parse("allTo@example.org"));
-    public static final List<Address> ALL_CC_ADDRESSES = Arrays.asList(Address.parse("allCc@example.org"));
+    private static final ReplyToAddresses TO_ADDRESSES = new ReplyToAddresses(Address.parse("to@example.org"));
+    private static final List<Address> ALL_TO_ADDRESSES = Arrays.asList(Address.parse("allTo@example.org"));
+    private static final List<Address> ALL_CC_ADDRESSES = Arrays.asList(Address.parse("allCc@example.org"));
```
Comment (cketti) [inline k9mail/src/test/java/com/fsck/k9/activity/compose/RecipientPresenterTest.java]: :+1:

File: k9mail/src/test/java/org/openintents/openpgp/util/ShadowOpenPgpAsyncTask.java (hunk 1 above)
Comment (philipwhiuk) [inline k9mail/src/test/java/org/openintents/openpgp/util/ShadowOpenPgpAsyncTask.java]: Shadows are basically like mocks that stub out bits of Android and allow you to test without a device. But they don't always do what you want. In this case, in order to simulate tasks being run on the executor you have to replace the shadow implementation (this specific case is covered here: http://stackoverflow.com/questions/26290867/asynctask-test-doesnt-execute ).

File: k9mail/src/main/java/com/fsck/k9/activity/MessageCompose.java [hunk 4]
```diff
@@ -1497,7 +1497,8 @@ public void onMessageBuildSuccess(MimeMessage message, boolean isDraft) {
                 message.setUid(mMessageReference.getUid());
             }
 
-            boolean saveRemotely = recipientPresenter.isAllowSavingDraftRemotely();
+            // TODO more appropriate logic here? not sure
+            boolean saveRemotely = !recipientPresenter.getCurrentCryptoStatus().shouldUsePgpMessageBuilder();
```
Comment (philipwhiuk) [inline k9mail/src/main/java/com/fsck/k9/activity/MessageCompose.java]: Maybe CryptoStatus can have a 'willBeEncrypted()'? That's what you're really after, right?

Comment (Valodim) [inline k9mail/src/main/java/com/fsck/k9/activity/MessageCompose.java]: This is more of a "might" be encrypted. The thing here is, the logic that decides whether a draft is uploaded or not really shouldn't live in `CryptoStatus`.

Comment (philipwhiuk) [inline k9mail/src/main/java/com/fsck/k9/activity/MessageCompose.java]: I don't think it should live in the Activity either which is probably the issue here. It should really be in a controller, but let's leave that for another PR.
